internal/manifest/img/parse_test.go

```diff
@@ -47,9 +52,13 @@ func TestParse(t *testing.T) {
 	for _, testCase := range tests {
 		t.Run(testCase.name, func(t *testing.T) {
 			var moduleTemplateFromFile v1beta2.ModuleTemplate
-			builder.ReadComponentDescriptorFromFile(testCase.DescriptorSourceFile,
+			builder.ReadComponentDescriptorFromFile(testCase.descriptorSourceFile,
 				&moduleTemplateFromFile)
-			descriptor, err := provider.NewCachedDescriptorProvider().GetDescriptor(&moduleTemplateFromFile)
+			ocmi, err := ocmidentity.New("kyma-project.io/module/template-operator", testCase.descriptorVersion)
+			require.NoError(t, err)
+			descriptor, err := provider.NewCachedDescriptorProvider(
```

Comment:
I didn't get the point, for each test case, you create a FakeService specifically, then what's the meaning for that `stopped` field, each FakeService not sharing the state.

And even you need that `stopped` state, you current implementation is thread unsafe for concurrent tests, you need a mutext for update it's state.

Reply:
This stopped state is required to handle just one thing - testing if caching "works". There's been such existing check in one of the tests, so I wanted to keep this logic.
How does this work? CacheDescriptorProvider is now opaque. One doesn't know if the ComponentDescriptor is fetched from the cache, or from the underlying service.
By stopping the service I can test if the ComponentDescriptor is really fetched from the cache. If the cache is not used/empty, the call is propagated to a (now stopped) service, and the panic there will ensure test failure.

Your remark about concurrency may be valid, I need to double-check.